pylot/localization/localization_operator.py

```diff
@@ -0,0 +1,234 @@
+import erdos
+from pylot.utils import Quaternion, Pose, Transform, Location, \
+        Vector3D, Rotation
+
+import numpy as np
+
+from functools import partial
+from collections import deque
+
+
+class LocalizationOperator(erdos.Operator):
+    def __init__(self, imu_stream, gnss_stream, ground_pose_stream,
+                 pose_stream, flags):
+        # Queue of saved messages.
+        self._imu_updates = deque()
+        self._gnss_updates = deque()
+        self._ground_pose_updates = None
+
+        # Streams to register a watermark on.
+        watermark_streams = []
+
+        # Register callbacks on both the IMU and GNSS streams.
+        imu_stream.add_callback(
+            partial(self.save, msg_type="IMU", queue=self._imu_updates))
+        watermark_streams.append(imu_stream)
+
+        gnss_stream.add_callback(
+            partial(self.save, msg_type="GNSS", queue=self._gnss_updates))
+        watermark_streams.append(gnss_stream)
+
+        # Register the ground pose stream, if provided.
+        if ground_pose_stream:
```

Comment:
The if statement is superfluous because ground_pose_stream will always be a stream. We can't pass None.